include/envoy/server/listener_manager.h

```diff
@@ -167,6 +167,14 @@ class ListenerManager {
    */
   virtual std::vector<std::reference_wrapper<Network::ListenerConfig>> listeners() PURE;
 
+  /**
+   * @return std::vector<std::reference_wrapper<Network::ListenerConfig>> a list of all currently
+   * known listeners, both active and warming. Note that this routine returns references to the
+   * existing listeners. The references are only valid in the context of the current call stack and
+   * should not be stored.
+   */
+  virtual std::vector<std::reference_wrapper<Network::ListenerConfig>> allListeners() PURE;
```

Comment:
When changing the API of a class, think about how the methods look in code that uses the class.  Do the names of functions make it clear what is happening?

In this case, the class has two public methods:

  `listeners()` : Returns some of the listeners.
  `allListeners()`: Returns all of the listeners.

If I saw a call to `listeners()` in some code, I would not expect that it returns some listeners, or know which ones.

Ideas to fix this:

1) Rename `listeners()` to `loadedListners()`, `activeListners()`, etc.

2) Make the type of listeners to return an enum:

```
  enum ListenerState {
     ACTIVE = 1,
     WARMING = 2,
     ALL = 3
  }
```

Function `listeners(ListenerState state)` would return the listeners in the given state.

Reply:
Thanks for the feedback.  I was thinking of changing `listeners()` to `activeListeners()` but I didn't want to be too disruptive to the code base since `listeners()` is used in a lot of different places (mostly in tests).  But if you're happy enough with this I can go ahead and do it.

Reply:
@asraa knows this code better than I do, and I would defer to her judgement.

Reply:
(2) Along with draining listeners might be good. It can default to active listeners as currently used. @lambdai What do you think?

Reply:
I am fine to add draining listener in the query. 
Though to fix the fuzzer issue we don't need the names in drain list.

Reply:
Sure, I'll do this and add draining listeners too